geemusic/utils/music_queue.py

```diff
@@ -46,6 +46,8 @@ def reset(self, tracks=[]):
                 song_id = track['trackId']
             elif 'track' in track:
                 song_id = track['track']['storeId']
+            elif 'u' in track:
+                song_id = track
```

Comment:
If you can move your shuffle implementation over to this class then I don't think this will be necessary.